### DISCUSSION TITLE
bug(syntax): Syntax Problem between Freeform and Fixedform

### DISCUSSION THREAD

**Discussion by zhongxiang117** [krvajal/vscode-fortran-support#1145] [Q&A - deprecated see Fortran Discourse]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I noticed that "the configuration of the selection language mode" does not work between `FortranFreeform` and `FortranFixedform`, if the unusual file extension is used.

For my seeing, the output info for both of them are:

```
["INFO" - 11:25:36 AM] using linter: gfortran located in: /usr/bin/gfortran
["INFO" - 11:25:36 AM] Linter.arguments:
-Wall
-ffree-line-length-none
-ffixed-line-length-none
```
It is not correct, instead, for `FortranFreeform`, the `Linter.arguments` should be `-ffree-form`.

### Screenshots

![image](https://github.com/user-attachments/assets/00df170b-7fcb-43e0-a643-55a1ff6dc6df)

### Code Example

```Fortran
! file should be with extension `.fpp` or any others, but not `.f90`

! shoud be freeform
program test
    print*, 'good'
end
```

### Fortran Version (Free/Fixed Form)

Free Form e.g. F90+

### Version of Modern Fortran

v3.2.0

### Version of Visual Studio Code

v1.85.0

### Platform and Architecture

Linux (Ubuntu)

### Additional Information

gfortran version v9.4.0 & v11.4.0

**Possible Solution:**

Add "builtin" argument `-ffree-form` when language mode `FortranFreeform` is selected.

![image](https://github.com/user-attachments/assets/602c6133-c5cc-40a6-be2a-bce1a62861d0)

**Comment by gnikit** [comment]
This is a linting issue. To make better sense of it, use the pre-release and provide the compiler version. `fpp` is registered as `FreeFormFortran` so I suspect you are running in an already resolved bug from the old release (you might have noticed I haven't made stable release in a while, completely unrelated to the stability of the pre-release and more related with admin things that are still pending!)

**Comment by zhongxiang117** [comment]
Sorry, let me firstly confirm this, the “linting issue”, is the linter builtin inside compiler? If so, I have the settings like:

```
"fortran.linter.compiler": "gfortran",
"fortran.linter.compilerPath": "/home/xiang/Applications/miniconda3core/envs/xutils/bin/gfortran",
```

then please have a look on following screenshot,

![DeepinScreenshot_select-area_20240920095227](https://github.com/user-attachments/assets/525413d8-1413-4d58-8014-2c1310855aa3)

As you will see, it still does not work at all.

Or you are mentioning different things, the linter, where can I find or compile it (cause I really don’t know whether it is a different thing rather than compiler itself)?
